eng/common/TestResources/New-TestResources.ps1

```diff
@@ -414,6 +412,19 @@ try {
         exit
     }
 
+    $templateFile = ""
```

Comment:
Instead of this, why don't we just use an ancestor search like [this](https://github.com/heaths/profile/blob/0ac9475b577218076bc3e792564edf553f23b02a/Modules/MyPrompt/MyPrompt.psm1#L91)? Seems much less cumbersome than splitting on directory paths. I mean, in this case, isn't the `$root` above potentially a package (to support _sdk/{serviceDirectory}/{package}_ so searching up would be more accurate than searching down to find one that matches?

Reply:
@heaths It's a fair question. I was in the mindset of adapting to the outputs from the existing code but should have considered this approach.

A challenge is that we don't know the sub directory, just the root, so we're searching from the parent anyways. I could write a depth first traversal that just returns on the first template file. I guess a downside of either implementation is in the following layout we would not know which template to pick, but I think supporting said layout isn't really a convention we should support anyways.

```
ServiceDirectory: foo

foo/bar/test-resources.json
foo/qux/test-resources.json
```

Reply:
In this scenario, going top-down makes sense then, but then how do we know which ones to pick up for a package? You said we still start from the service directory, but how are we picking the right package directory (if any) test-resources.jsons now?

Reply:
Updated to throw